keras/layers/wrappers.py

```diff
@@ -163,30 +163,47 @@ def compute_output_shape(self, input_shape):
         child_input_shape = (input_shape[0],) + input_shape[2:]
         child_output_shape = self.layer.compute_output_shape(child_input_shape)
         timesteps = input_shape[1]
-        return (child_output_shape[0], timesteps) + child_output_shape[1:]
+        if isinstance(child_output_shape, tuple):
+            return (child_output_shape[0], timesteps) + child_output_shape[1:]
+        elif isinstance(child_output_shape, list):
+            return [(shape[0], timesteps) + shape[1:] for shape in child_output_shape]
+        msg = 'output shape should be a tuple (single output) or a list (multiple outputs)'
+        raise ValueError(msg)
 
     def call(self, inputs, training=None, mask=None):
+        class LPWrapper:
+            # hack equivalent to nonlocal keyword
+            # This design was chosen for Python 2.7 compatibility.
+            uses_learning_phase = False
+
         kwargs = {}
         if has_arg(self.layer.call, 'training'):
             kwargs['training'] = training
-        uses_learning_phase = False
 
         input_shape = K.int_shape(inputs)
         if input_shape[0]:
             # batch size matters, use rnn-based implementation
             def step(x, _):
-                global uses_learning_phase
                 output = self.layer.call(x, **kwargs)
-                if hasattr(output, '_uses_learning_phase'):
-                    uses_learning_phase = (output._uses_learning_phase or
-                                           uses_learning_phase)
+                if not isinstance(output, list):
+                    output = [output]
+                uses_learning_phase = []
```

Comment:
Do we not need a boolean for each output though? If not then yeah a list is not necessary but `_uses_learning_phase` is a tensor attribute so what should [this](https://github.com/PavlosMelissinos/keras/blob/17e7e612773555d2fa79fdcfe305b0a993ef4eb7/keras/layers/wrappers.py#L246-L248) be replaced with?

here's a code snippet of the part in question:
```
for idx, yi in enumerate(y):
    if LPWrapper.uses_learning_phase[idx]:
        yi._uses_learning_phase = True
```

In my understanding, the point of the global `uses_learning_phase` variable was to transfer the old value to the converted output. This is the one produced [on line 208 in the master branch of keras](https://github.com/fchollet/keras/blob/master/keras/layers/wrappers.py#L208), which I've modified to become
```
y = [K.reshape(yi, (-1, input_length) + output_shape[idx][2:])
for idx, yi in enumerate(y)])
```
What am I missing?